tests/pause_workflow_execution_test.go

```diff
@@ -112,17 +113,93 @@ func (s *PauseWorkflowExecutionSuite) TestPauseWorkflowExecution() {
-func (s *PauseWorkflowExecutionSuite) TestPauseWorkflowExecutionFailsWhenDisabled() {
+// TestPauseWorkflowExecutionRequestValidation tests that pause workflow execution request validation. We don't really need a valid workflow to test this.
+// - fails when the identity is too long.
+// - fails when the reason is too long.
+// - fails when the request id is too long.
+// - fails when the dynamic config is disabled.
+func (s *PauseWorkflowExecutionSuite) TestPauseWorkflowExecutionRequestValidation() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	namespaceName := s.Namespace().String()
+
+	// fails when the identity is too long.
+	pauseRequest := &workflowservice.PauseWorkflowExecutionRequest{
+		Namespace:  namespaceName,
+		WorkflowId: "test-workflow-id",
+		RunId:      uuid.New(),
+		Identity:   strings.Repeat("x", 2000),
+		Reason:     s.pauseReason,
+		RequestId:  uuid.New(),
+	}
+	resp, err := s.FrontendClient().PauseWorkflowExecution(ctx, pauseRequest)
+	s.Error(err)
+	s.Nil(resp)
+	var invalidArgumentErr *serviceerror.InvalidArgument
+	s.ErrorAs(err, &invalidArgumentErr)
+	s.NotNil(invalidArgumentErr)
+	s.Contains(invalidArgumentErr.Error(), "identity is too long.")
+
+	// fails when the reason is too long.
+	pauseRequest = &workflowservice.PauseWorkflowExecutionRequest{
+		Namespace:  namespaceName,
+		WorkflowId: "test-workflow-id",
+		RunId:      uuid.New(),
+		Identity:   s.pauseIdentity,
+		Reason:     strings.Repeat("x", 2000),
+		RequestId:  uuid.New(),
+	}
+	resp, err = s.FrontendClient().PauseWorkflowExecution(ctx, pauseRequest)
+	s.Error(err)
+	s.Nil(resp)
+	s.ErrorAs(err, &invalidArgumentErr)
+	s.NotNil(invalidArgumentErr)
+	s.Contains(invalidArgumentErr.Error(), "reason is too long.")
+
+	// fails when the request id is too long.
+	pauseRequest = &workflowservice.PauseWorkflowExecutionRequest{
+		Namespace:  namespaceName,
+		WorkflowId: "test-workflow-id",
+		RunId:      uuid.New(),
+		Identity:   s.pauseIdentity,
+		Reason:     s.pauseReason,
+		RequestId:  strings.Repeat("x", 2000),
+	}
+	resp, err = s.FrontendClient().PauseWorkflowExecution(ctx, pauseRequest)
+	s.Error(err)
+	s.Nil(resp)
+	s.ErrorAs(err, &invalidArgumentErr)
+	s.NotNil(invalidArgumentErr)
+	s.Contains(invalidArgumentErr.Error(), "request id is too long.")
+
+	// fails when the dynamic config is disabled.
 	s.OverrideDynamicConfig(dynamicconfig.WorkflowPauseEnabled, false)
+	pauseRequest = &workflowservice.PauseWorkflowExecutionRequest{
+		Namespace:  namespaceName,
+		WorkflowId: "test-workflow-id",
+		RunId:      uuid.New(),
+		Identity:   s.pauseIdentity,
+		Reason:     s.pauseReason,
+		RequestId:  uuid.New(),
+	}
+	resp, err = s.FrontendClient().PauseWorkflowExecution(ctx, pauseRequest)
+	s.Error(err)
+	s.Nil(resp)
+	var unimplementedErr *serviceerror.Unimplemented
+	s.ErrorAs(err, &unimplementedErr)
```

Comment:
Is `Unimplemented` always the error we return for a feature that exists but isn't turned on for a namespace? That seems unintuitive but a reasonable response.